### PR TITLE
Bug 2089419: Don't block upgrades when another CSI driver is found

### PR DESCRIPTION
**Pull request by jsafrane**
The first commit refactors two bools into enum, to be able to add a third possible outcome of a failed check.
The second commit then blocks CSI driver installation when the driver is already installed and it's not the OCP one. The operator looks "Available: true" + "Upgradeable: true", however, the Upgradeable condition has a message `found existing unsupported csi.vsphere.vmware.com driver`.

WIP:
* Add unit test.
* Think about useful alert.

@openshift/storage

### CODE REVIEW

File: pkg/operator/vspherecontroller/checks/check_error.go [hunk 1]
```diff
@@ -124,6 +147,11 @@ func CheckClusterStatus(result ClusterCheckResult, apiDependencies KubeAPIInterf
 		}
 
 		return ClusterCheckBlockUpgrade, result
+
+	case CheckActionBlockInstall:
+		return ClusterCheckBlockInstall, result
```
Comment (gnufied) [inline pkg/operator/vspherecontroller/checks/check_error.go:152]: `ClusterCheckBlockInstall` should may be renamed to `ClusterCheckBlockDriverInstall`. I know it is bit mountful but perhaps more accurate.

Comment (gnufied) [inline pkg/operator/vspherecontroller/checks/check_error.go:152]: `ClusterCheckBlockInstall` should may be renamed to `ClusterCheckBlockDriverInstall`. I know it is bit mountful but perhaps more accurate.

Comment (jsafrane) [reply pkg/operator/vspherecontroller/checks/check_error.go:152]: done

Comment (jsafrane) [reply pkg/operator/vspherecontroller/checks/check_error.go:152]: done

**Comment by jsafrane** [comment]
/retest
installation flakes

**Pull request merged**